src/audio/codec_adapter/codec/cadence.c

```diff
@@ -464,6 +492,10 @@ static int cadence_codec_prepare(struct processing_module *mod)
 
 	comp_dbg(dev, "cadence_codec_prepare() start");
 
+	ret = cadence_codec_post_init(mod);
```

Comment:
could we be leaking here if we are calling this multiple times ?

Reply:
@lgirdwood yes, you are right. Latest update should take care of that.